downloads/swan-lake-release-notes/swan-lake-alpha3/RELEASE_NOTE.md

```diff
@@ -2,41 +2,928 @@
+
+Also, these variable declarations cannot contain an `isolated` or `configurable` qualifier.
+
+```ballerina
+type Person record {|
+    string name;
+    boolean married;
+|};
+
+function getList() returns [int, float] => [1, 2.5];
+
+function getPerson() returns Person => {name: "John", married: true};
+
+function getError() returns error => error("error message", code = 1001, fatal = true);
+
+// Module-level variable declaration with a list binding pattern. 
+[int, float] [a, b] = getList();
+
+// Module-level variable declaration with a mapping binding pattern.
+Person {name: firstName, married: isMarried} = getPerson();
+
+// Module-level variable declaration with an error binding pattern.
+error error(message, code = errCode) = getError();
+```
+
+##### Support for Module-level Public Variables
+
+Module-level variables can now be declared as public using the `public` qualifier. Such variables will be visible outside the modules in which they are declared.
+
+Isolated variables and variables declared with `var` cannot be declared as public variables.
+
+```ballerina
+public string name = "Ballerina";
+
+public [int, float] [a, b] = [1, 2.5];
+```
+
+##### Improvement to Annotation Attachment with Empty Mapping Constructor Expression
+
+If the type of the annotation is a mapping type for which an empty mapping constructor is valid, the mapping constructor expression is no longer mandatory in the annotation attachment.
+
+The absence of the mapping constructor expression in such an annotation attachment is equivalent to specifying a mapping constructor expression with no fields.
+```ballerina
+type Annot record {|
+    int[] i = [];
+|};
+
+public annotation Annot v1 on function;
+
+@v1 // Same as `@v1 {}`
+public function main() {
+}
+```
+
+##### Introduction of the `function` Function Type Descriptor to Represent Any Function
+
+A new `function` type descriptor has been introduced to represent all function values.
+
+```ballerina
+import ballerina/io;
+
+function add(int v1, int v2) returns int => v1 + v2;
+
+function compare(int v1, int v2) returns boolean => v1 < v2;
+
+public function main() {
+    // A variable of type `function` can hold any function value.
+    function f = add;
+    io:println("Process (add, 1, 2): ", process(f, 1, 2)); // Prints `Process (add, 1, 2): 3`
+    io:println("Process (compare, 1, 2): ", process(compare, 1, 2)); // Prints `Process (compare, 1, 2): 0`
+}
+
+function process(function func, int v1, int v2) returns int {
+    // A function of type `function` cannot be called directly.
+    // A function value assigned to a `function`-typed variable
+    // can only be called after the type is narrowed to the relevant type.
+    if (func is function (int, int) returns int) {
+        return func(v1, v2);
+    }
+    return 0;
+}
+```
+
+##### Improved lang library functions
```

Comment:
This says "Improved" while a section before this said "Improvement". Shouldn't all these be in the same format? Also, shouldn't the title case be followed here as well?

Reply:
Shall we say

```suggestion
##### New Lang Library Functions
```

We only have one though.

Reply:
Fixed